File: Question2.cpp
#include <iostream>
#include <string>
#include <cctype>
using namespace std;

bool CheckPalindrome(string Text) {
    
    string cleanedText;
    
    // Convert characters to lowercase and keep only alphabetical characters
    for (char c : Text) {
        if (isalpha(c)) {
            cleanedText += tolower(c);
        }
    }
    
    // Check if the cleaned text is equal to its reverse
    return cleanedText == string(cleanedText.rbegin(), cleanedText.rend());
}

int main() {
    string userInput;
    
    cout << "Enter a string: ";
    getline(cin, userInput);
    
    if (CheckPalindrome(userInput)) {
        cout << "Yes, it's a palindrome!" << endl;
    } else {
        cout << "No, it's not a palindrome." << endl;
    }
    
    return 0;
}

File: Question2.py
def CheckPalindrome(Text):
    Text = ''.join(char.lower() for char in Text if char.isalpha())
    
    #.isalpha() checks if all the characters in a string are alphabetical characters
    #converts the new string to lowercase using the lower() method.
    
    return Text == Text[::-1]
#reverses the order of elements in a sequence

def main():
    user_input = input("Enter a string: ")
    
    if CheckPalindrome(user_input):
        print("Yes, it's a palindrome!")
        
    else:
        print("No, it's not a palindrome.")

if __name__ == "__main__":
    main()

